src/api.jl

```diff
@@ -619,6 +619,33 @@ function get_transaction(ctx::Context, id::AbstractString; kw...)
     return rsp.transaction
 end
 
+function get_transaction_events(ctx::Context, txn_id::AbstractString; kw...)
+    # TODO: support a cancellation token
+    path = PATH_ASYNC_TRANSACTIONS * "/$txn_id/events"
+    continuation_token = ""
+    events = Channel()
+    @spawn begin
+        try
```

Comment:
@vilterp the more idiomatic way to spawn a task for the channel and to close the channel if there's an error is with the `spawn=true` do-constructor:
```julia
    events = Channel(spawn=true) do
```
Then you can remove your catch-block at the end.

Reply:
Cool, thanks. What do you think would be a good idiom for allowing this to be cancelled by the caller?

Reply:
Yeah, would be good to open source that and use it here. Prob will go without for the time being

Reply:
I think you can leave a note in the docstring that they should close the channel to cancel the task.

Reply:
Hm, that's gonna throw an error when it tries to `push!` an event onto it and it's closed tho 🤔 

I do think cancellation signal should be something passed in as an argument for that purpose